src/core/interfaces.ts

```diff
@@ -88,11 +88,22 @@ export type Bounds = {
 };
 
 /**
+ * Client bounds
+ */
+export interface IEntityBounds {
+  x: number;
+  y: number;
+  width: number;
+  height: number;
+}
+
+/**
  * An object representing a data-backed visual entity inside a Component.
  */
 export interface IEntity<C extends Component> {
+  bounds?: IEntityBounds; // TODO make non-optional
```

Comment:
this is very close to the resolved attrs, just a subset including "x", "y", "width", and "height". It's used for .entityNearest and .entityIntersecting for tooltips and selections.  It is specific to entities, which is why it's here.